pkg/skaffold/runner/deploy_test.go

```diff
@@ -133,3 +135,26 @@ func TestDeployNamespace(t *testing.T) {
 		})
 	}
 }
+
+func TestSkaffoldDeployRenderOnly(t *testing.T) {
+	testutil.Run(t, "does not make kubectl calls", func(t *testutil.T) {
+		runCtx := &runcontext.RunContext{
+			Opts: config.SkaffoldOptions{
+				Namespace:  "testNamespace",
+				RenderOnly: true,
+			},
+			KubeContext: "kubecontext",
+		}
+
+		r := SkaffoldRunner{
+			runCtx:     runCtx,
+			kubectlCLI: kubectl.NewFromRunContext(runCtx),
+			deployer:   getDeployer(runCtx),
+		}
+		var builds []build.Artifact
+
+		err := r.Deploy(context.Background(), ioutil.Discard, builds)
```

Comment:
@kissmikijr Thanks you rebasing and addressing the comments. Can you please explain how does this test ensure your are not making any kubectl calls?
Does this test point to a non existing `kubecontext` and hence should fail.

Reply:
if that is the case, can you please make it explicit by either adding a comment or 

```
KubeContext: "does-not-exist",
```

Reply:
added `KubeContext: "does-not-exist"`